### PR TITLE
feat(nn4k): implement openai invoker and local hf executor

### PR DESCRIPTION
**Pull request by xionghuaidong**

### CODE REVIEW

File: python/nn4k/nn4k/utils/class_importing.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+import importlib
+from typing import Tuple
+
+
+def split_module_class_name(name: str, text: str) -> Tuple[str, str]:
+    """
+    Split `name` as module name and class name pair.
+
+    :param name: fully qualified class name, e.g. ``foo.bar.MyClass``
```
Comment (didicout) [inline python/nn4k/nn4k/utils/class_importing.py:20]: pls unify the doc-string style

**Review by didicout — CHANGES_REQUESTED** [review]

File: python/nn4k/nn4k/utils/class_importing.py (hunk 1 above)
Comment (xionghuaidong) [reply python/nn4k/nn4k/utils/class_importing.py:20]: done

File: python/nn4k/nn4k/executor/hugging_face.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        else:
+            max_input_length = 1024
+        if "max_output_length" in kwargs:
+            max_output_length = kwargs.pop("max_output_length")
+        else:
+            max_output_length = 1024
+        if "do_sample" in kwargs:
+            do_sample = kwargs.pop("do_sample")
+        else:
+            do_sample = False
```
[the remark this answers is not in the document]
Comment (xionghuaidong) [reply python/nn4k/nn4k/executor/hugging_face.py]: done

File: python/nn4k/nn4k/executor/hugging_face.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        ).to(model.device)
+        output_ids = model.generate(
+            **input_ids,
+            max_new_tokens=max_output_length,
+            do_sample=do_sample,
+            eos_token_id=tokenizer.eos_token_id,
+            pad_token_id=tokenizer.pad_token_id,
+            **kwargs,
+        )
+        outputs = [
```
[the remark this answers is not in the document]
Comment (xionghuaidong) [reply python/nn4k/nn4k/executor/hugging_face.py:98]: done

File: python/nn4k/nn4k/invoker/base.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        :rtype: NNInvoker
+        :raises RuntimeError: if the NN config is not recognized
+        """
+        from nn4k.nnhub import NNHub
+        from nn4k.utils.config_parsing import preprocess_config
+        from nn4k.utils.config_parsing import get_string_field
+        from nn4k.utils.class_importing import dynamic_import_class
+
+        nn_config = preprocess_config(nn_config)
+        nn_invoker = nn_config.get("nn_invoker")
```
[the remark this answers is not in the document]
Comment (xionghuaidong) [reply python/nn4k/nn4k/invoker/base.py]: done

File: python/nn4k/nn4k/invoker/base.py [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+                raise RuntimeError(message)
+            invoker = invoker_class.from_config(nn_config)
+            return invoker
+
+        hub = NNHub.get_instance()
+        invoker = hub.get_invoker(nn_config)
+        if invoker is not None:
+            return invoker
+
+        nn_name = nn_config.get("nn_name")
```
[the remark this answers is not in the document]
Comment (xionghuaidong) [reply python/nn4k/nn4k/invoker/base.py]: done

File: python/nn4k/nn4k/invoker/base.py [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+        if nn_version is not None:
+            nn_version = get_string_field(nn_config, "nn_version", "NN model version")
+        message = "can not create invoker for NN config"
+        if nn_name is not None:
+            message += "; model: %r" % nn_name
+            if nn_version is not None:
+                message += ", version: %r" % nn_version
+        raise RuntimeError(message)
+
+    def submit_inference(self, submit_mode="k8s"):
```
[the remark this answers is not in the document]
Comment (xionghuaidong) [reply python/nn4k/nn4k/invoker/base.py]: done

File: python/nn4k/nn4k/invoker/base.py [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+        return self._nn_executor.inference(data, **kwargs)
+
+    def warmup_local_model(self):
+        """
+        Implement local model warming up logic for local invoker.
+        """
+        from nn4k.nnhub import NNHub
+        from nn4k.utils.config_parsing import get_string_field
+
+        nn_name = get_string_field(self.init_args, "nn_name", "NN model name")
```
[the remark this answers is not in the document]
Comment (xionghuaidong) [reply python/nn4k/nn4k/invoker/base.py]: done

File: python/nn4k/setup.py [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+__version__ = "{version}"
+"""
+    wf.write(content)
+
+setup(
+    name=package_name,
+    version=version,
+    description="nn4k",
+    author="xionghuaidong",
+    author_email="huaidong.xhd@antgroup.com",
```
[the remark this answers is not in the document]
Comment (xionghuaidong) [reply python/nn4k/setup.py]: fixed

**Review by leywar — APPROVED** [review]

**Review by baifuyu — APPROVED** [review]
LGTM

**Pull request merged**